tests/e2e/default/navigation/navigation.wdio-spec.js

```diff
@@ -1,34 +1,83 @@
+const userFactory = require('../../../factories/cht/users/users');
+const placeFactory = require('../../../factories/cht/contacts/place');
 
 describe('Navigation tests', async () => {
-  beforeEach(async () => {
-    await loginPage.cookieLogin();
-  });
+  describe('Navigation functionality', async () => {
+    beforeEach(async () => {
+      await loginPage.cookieLogin();
+    });
 
-  it('should open Messages tab', async () => {
-    await commonPage.goToMessages();
-    expect(await commonPage.isMessagesListPresent());
-  });
+    after(async () => {
+      await commonPage.logout();
+    });
 
-  it('should open tasks tab', async () => {
-    await commonPage.goToTasks();
-    expect(await commonPage.isTasksListPresent());
-  });
+    it('should open Messages tab', async () => {
+      await commonPage.goToMessages();
+      expect(await commonPage.isMessagesListPresent());
+    });
 
-  it('should open Reports or History tab', async () => {
-    await commonPage.goToReports();
-    expect(await commonPage.isReportsListPresent());
-  });
+    it('should open tasks tab', async () => {
+      await commonPage.goToTasks();
+      expect(await commonPage.isTasksListPresent());
+    });
+
+    it('should open Reports or History tab', async () => {
+      await commonPage.goToReports();
+      expect(await commonPage.isReportsListPresent());
+    });
+
+    it('should open Contacts or Peoples tab', async () => {
+      await commonPage.goToPeople();
+      expect(await commonPage.isPeopleListPresent());
+    });
 
-  it('should open Contacts or Peoples tab', async () => {
-    await commonPage.goToPeople();
-    expect(await commonPage.isPeopleListPresent());
+    it('should open Analytics tab', async () => {
+      await commonPage.goToAnalytics();
+      expect(await commonPage.isTargetMenuItemPresent());
+      expect(await commonPage.isTargetAggregatesMenuItemPresent());
+    });
   });
 
-  it('should open Analytics tab', async () => {
-    await commonPage.goToAnalytics();
-    expect(await commonPage.isTargetMenuItemPresent());
-    expect(await commonPage.isTargetAggregatesMenuItemPresent());
+  describe('Navigation view', async () => {
+    const places = placeFactory.generateHierarchy();
+    const districtHospital = places.get('district_hospital');
+    const user = userFactory.build({ place: districtHospital._id, roles: ['program_officer'] });
+
+    beforeEach(async () => {
+      await utils.saveDocs([...places.values()]);
+      await utils.createUsers([user]);
+    });
+
+    afterEach(async () => {
+      await commonPage.logout();
+      await utils.deleteUsers([user]);
+      await utils.revertDb([/^form:/], true);
+      await utils.revertSettings(true);
+    });
+
+    it('should display tab labels, when all tabs are enabled', async () => {
+      await loginPage.cookieLogin();
+      const tabsButtonLabelsNames = await commonPage.getAllButtonLabelsNames();
+      expect(tabsButtonLabelsNames.length).to.deep.equal(5);
```

Comment:
If I remember correctly (I haven't tried ), since it's already checking `.to.deep.equal(` in the following line, the array length is already considered. Please feel free to remove this line.  
Same with the test below